### PR TITLE
190608 bug

### PR DESCRIPTION
**Pull request by SunilSimha**
**Bug fixes:**
- `DM_IGM` units fixed from `pc/cm**2` to `pc/cm**3`.
- Default `neval` made consistent across `average_DMIGM` and `avg_DMHalos`.
- DM evaluations are now based on `Planck15` by default.
- `frac_in_halos` is now computed between the same mass limits as in `avg_DMhalos`.
- The default `average_DMIGM` added to the default `avg_DMHalos` now sums up to the default `average_DM`.
- `average_DMIGM` and `average_DMhalos` are both functions of `f_hot` now.
-DM_halos takes `f_diffuse(z)` into account now.
-Renamed `avg_DMHalos` to `average_DMHalos`.

**New functions:**
- `f_diffuse`: computes the cosmic mass fraction of baryons in diffuse phase.
- `ne_cosmic`: computes the average cosmic electron density as a function of z.

### CODE REVIEW

File: frb/igm.py [hunk 1]
```diff
@@ -153,8 +153,70 @@ def z_from_DM(DM, cosmo=None, coord=None, corr_nuisance=True):
...
+  number density as a function of redshift.
+  Args:
+    z: float or ndarray
+      Redshift
+    cosmo: astropy Cosmology, optional
+      Cosmology in which the calculations
+      are to be performed.
+  Returns:
+    ne_cosmic: astropy Quantity
+      Average number density of
```
Comment (profxj) [inline frb/igm.py]: comoving or physical?

File: frb/igm.py [hunk 2]
```diff
@@ -153,8 +153,70 @@ def z_from_DM(DM, cosmo=None, coord=None, corr_nuisance=True):
...
+  # Diffuse gas fraction
+  f_diffuse = 1 - ((rho_Mstar+rho_ISM)/rho_b).value
+  if not return_rho:
+    return f_diffuse
+  else:
+    return f_diffuse, rho_b*f_diffuse*(1+z)**3
+
+def ne_cosmic(z, cosmo = Planck15, mu = 4/3):
+  """
+  Calculate the average cosmic electron
```
Comment (profxj) [inline frb/igm.py]: comoving or physical?

**Comment by profxj** [comment]
Also, you are going to need to update the tests.
These are now failing.

File: frb/igm.py (hunk 1 above)
Comment (SunilSimha) [reply frb/igm.py]: Physical. Just updated the docstring to reflect this,

**Comment by profxj** [comment]
Merging.

**Pull request merged**